### ISSUE TITLE
Add CI Testing of Live Share Integration

### ISSUE THREAD

**Issue by johnpapa** [johnpapa/vscode-peacock#148]
Let's continue the discussion of how to test the live share integration in CI here @legomushroom

Can we load the live share environment in CI?

Should we run the regular tests alone, then run a second set of tests for Live share? I kind of like this because it allows us to more easily test one set with live share installed, and the second set without it being there.

We also want to make sure that Peacock has tests to make sure that if live share is not installed that those features cannot be activated and cause an error for the user. I think we need to add those tests.

**Comment by legomushroom** [comment]
> Can we load the live share environment in CI?

Yes, we can install Live Share extension on the CI environment by adding an additional step to the build pipeline.

> Should we run the regular tests alone, then run a second set of tests for Live share? I kind of like this because it allows us to more easily test one set with live share installed, and the second set without it being there.

I imagine this:

	// ... all other build steps
	- Compile tests
	- Run "core" tests(`just-test`)
    - Step to install Live Share
	- Run "live share" tests (`just-live-share-test`)

This will allow us to run the tests before live share installed and then after the live share. I'll modify the live share tests to have the `authentication` step before each of them.

> We also want to make sure that Peacock has tests to make sure that if live share is not installed that those features cannot be activated and cause an error for the user. I think we need to add those tests.

Makes perfect sense to me, though I thought the `"core"` tests are kind of testing that already. If we had the build pipeline like outlined above, would it address your concerns?

Thanks,
Oleg

**Comment by johnpapa** [comment]
I think if we have both `npm test` and npm run test-live-share` as separate setps in the CI process, that would work.

**Comment by johnpapa** [comment]
This internal task has been open since 2019 and hasn't been prioritized. Closing to keep the issue tracker focused. If Live Share CI testing becomes a priority, we can open a fresh issue with updated requirements.
